### PR TITLE
Idempotent setAnimation()

### PR DESCRIPTION
**Pull request by islemaster**
In Gamelab, `Sprite.setAnimation(label)` should be a no-op if the requested animation is already the sprite's current animation.

If the requested animation is different, the behavior is the same: We switch to the new animation, set it to the first frame, and start playing it.

Most of this PR is new unit tests covering the behavior of `setAnimation`.

### CODE REVIEW

File: apps/test/util/gamelab/TestableGameLabP5.js [hunk 1]
```diff
@@ -46,3 +52,33 @@ function createStatefulInterpreter() {
...
+  // Not identical
+  expect(sheet1).not.to.equal(sheet2);
+  // But frame array is same size
+  // Note: Would check deep equal but cloning frames adds 'name' property in
+  //       some cases
+  expect(sheet1.frames.length).to.equal(sheet2.frames.length);
+  // And other props exactly equal
+  expect(sheet1.image).to.equal(sheet2.image);
+  expect(sheet1.frame_width).to.equal(sheet2.frame_width);
+  expect(sheet1.frame_height).to.equal(sheet2.frame_height);
```
Comment (caleybrock) [inline apps/test/util/gamelab/TestableGameLabP5.js:81]: duplicate check here

File: apps/test/unit/gamelab/GameLabSpriteTest.js [hunk 2]
```diff
@@ -300,6 +303,86 @@ describe('GameLabSprite', function () {
...
+      expect(sprite.animation.getFrame()).to.equal(0);
+      expect(sprite.animation.playing).to.be.true;
+      sprite.animation.nextFrame();
+      sprite.animation.nextFrame();
+      sprite.animation.nextFrame();
+      expect(sprite.animation.getFrame()).to.equal(3);
+      sprite.pause();
+      expect(sprite.animation.playing).to.be.false;
+
+      // Set second animation, observe that we are back on first frame
```
Comment (caleybrock) [inline apps/test/unit/gamelab/GameLabSpriteTest.js]: I think this comment is probably copied from above and is irrelevant. Also isn't this the case where we would want the animation to be playing, and continue playing but not reset? Wouldn't be a bad idea to test both cases.

File: apps/test/util/gamelab/TestableGameLabP5.js (hunk 1 above)
Comment (islemaster) [reply apps/test/util/gamelab/TestableGameLabP5.js:81]: Whoops!  Nice catch, thanks!

**Comment by islemaster** [comment]
Redid tests based on feedback, PTAL.

**Comment by caleybrock** [comment]
👀

**Comment by caleybrock** [comment]
awesome, thanks!

**Comment by islemaster** [comment]
UI test failures should be unrelated to this change, but re-running Circle anyway.

**Pull request merged**